walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,95 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    //input: events collection and request for meeting
+    //output: collection of all times the proposed meeting could be held at
+
+    Collection<String> attendeesList = request.getAttendees();
+    long meetingDuration = request.getDuration();
+    List<TimeRange> proposedTimeRanges = new ArrayList<TimeRange>();
+    List<TimeRange> clashAttendeeEvents = new ArrayList<TimeRange>();
+
+    int startDay = TimeRange.START_OF_DAY;
+    int endDay = TimeRange.END_OF_DAY;
+
+    /*loop through all the planned events and find the relevant ones 
+    (where there's an attendee clash)
+    */
+    for (Event event: events) {
+    //get the attendees from the specific event you're checking the attendees of
+      Set<String> eventAttendees = event.getAttendees();
+      for (String attendee : attendeesList) {
+        if (eventAttendees.contains(attendee)) {
+          clashAttendeeEvents.add(event.getWhen());
```

Comment:
nit: maybe you need to `break` after you add this event time range? just to avoid having it as many times as the number of conflicts (e.g. one event is attended by two persons that are also part of the `attendeesList`). It should be fine though as you already handle this below, it's just easier to read/debug :)

Reply:
Thanks Mostafa!